flow/probes/ebpf.go

```diff
@@ -110,19 +162,52 @@ func (p *EBPFProbe) newFlowOperation(ebpfFlow *EBPFFlow, kernFlow *C.struct_flow
 		B:        net.HardwareAddr(linkB).String(),
 		ID:       int64(kernFlow.link_layer.id),
 	}
-	f.LayersPath = "Ethernet"
-
-	nVLAN := uint8(kernFlow.link_layer.vlans)
-	for i := uint8(0); i < nVLAN; i++ {
-		f.LayersPath += "/Dot1Q"
-	}
-
-	if layersFlag&uint8(C.ARP_LAYER) > 0 {
-		f.LayersPath += "/ARP"
+	f.LayersPath = kernLayersPath(kernFlow)
+
+	if strings.Contains(f.LayersPath, "GRE") {
+		layersPaths := strings.Split(f.LayersPath, "/GRE/")
```

Comment:
nit: use strings.Index(f.Layers.Path, "/GRE/") as it's faster and would handle the (very improbable) case where there are multiple GRE layers

Reply:
golang implementation : 

func Contains(s, substr string) bool {
	return Index(s, substr) >= 0
}

but ok

Reply:
I was talking about the `Split`, not the `Contain`